tests/main.test.ts

```diff
@@ -1,160 +1,183 @@
-import { drop } from "@mswjs/data";
 import { db } from "./__mocks__/db";
 import { server } from "./__mocks__/node";
-import { expect, describe, beforeAll, beforeEach, afterAll, afterEach, it } from "@jest/globals";
-import { Context } from "../src/types/context";
+import usersGet from "./__mocks__/users-get.json";
+import { describe, beforeAll, beforeEach, afterAll, afterEach, it, jest } from "@jest/globals";
+import { Context, SupportedEvents } from "../src/types";
+import { drop } from "@mswjs/data";
+import issueTemplate from "./__mocks__/issue-template";
+import repoTemplate from "./__mocks__/repo-template";
+import { logger } from "../src/helpers/errors";
 import { Octokit } from "@octokit/rest";
-import { STRINGS } from "./__mocks__/strings";
-import { createComment, setupTests } from "./__mocks__/helpers";
-import manifest from "../manifest.json";
-import dotenv from "dotenv";
-import { Logs } from "@ubiquity-dao/ubiquibot-logger";
-import { Env } from "../src/types";
-import { runPlugin } from "../src/plugin";
-
-dotenv.config();
-jest.requireActual("@octokit/rest");
-const octokit = new Octokit();
+import { CompletionsType } from "../src/adapters/claude/helpers/completions";
+import pullTemplate from "./__mocks__/pull-template";
+
+// Mock constants
+const MOCK_ANSWER_PASSED = "{confidenceThreshold: 1, reviewComment: 'passed'}";
+
+jest.unstable_mockModule("../src/helpers/pull-helpers/fetch-diff", () => ({
+  fetchPullRequestDiff: jest.fn(() => ({
+    diff: "abc",
+  })),
+}));
+
+beforeAll(() => {
+  server.listen();
+});
+
+afterEach(() => {
+  drop(db);
+  server.resetHandlers();
+});
+
+afterAll(() => server.close());
+
+jest.unstable_mockModule("../src/helpers/pull-helpers/fetch-diff", () => ({
+  fetchPullRequestDiff: jest.fn(() => ({
+    diff: "abc",
+  })),
+}));
 
 beforeAll(() => {
   server.listen();
 });
+
 afterEach(() => {
+  drop(db);
   server.resetHandlers();
-  jest.clearAllMocks();
 });
+
 afterAll(() => server.close());
 
-describe("Plugin tests", () => {
+describe("Pull Reviewer tests", () => {
   beforeEach(async () => {
-    drop(db);
+    jest.clearAllMocks();
     await setupTests();
   });
 
-  it("Should serve the manifest file", async () => {
-    const worker = (await import("../src/worker")).default;
-    const response = await worker.fetch(new Request("http://localhost/manifest"), {});
-    const content = await response.json();
-    expect(content).toEqual(manifest);
-  });
+  describe("Perform pull precheck", () => {
```

Comment:
try to create more tests like review can only be done again after 24 hours and other edge cases

Reply:
i'll need a day to add exhaustive tests